exp/internal/controllers/machinepool_controller_phases_test.go

```diff
@@ -1249,13 +1249,14 @@ func TestReconcileMachinePoolScaleToFromZero(t *testing.T) {
 		g.Expect(res.Requeue).To(BeFalse())
 
 		r.reconcilePhase(machinepool)
-
 		g.Expect(machinepool.Status.GetTypedPhase()).To(Equal(expv1.MachinePoolPhaseRunning))
 
 		delNode := &corev1.Node{}
-		err = env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(node), delNode)
-		g.Expect(err).ToNot(BeNil())
-		g.Expect(apierrors.IsNotFound(err)).To(BeTrue())
+		g.Eventually(func(g Gomega) {
+			err = fakeClient.Get(ctx, client.ObjectKeyFromObject(node), delNode)
```

Comment:
as far as I can tell the node is never created in the fakeClient. I think we can't switch from an env client to the fake client here

Reply:
Completely right - let me fix that.

/hold